gtkui: add copy/paste support

cboxdoerfer opened the pull request:
This is quite a big patch so I don't expect it to get merged right away.

What this PR does: It adds copy, cut and paste support to the ddblistview, the pltbrowser and the pltmenu. C&P works both from DeaDBeeF to file managers and the other way around, and of course within DeaDBeeF (copy track(s) in playlist1 and paste it/them in playlist2). Pasting tracks to the pltbrowser first creates a new playlist and then pastes the tracks there. C&P can be triggered either via menu items or via shortcuts (ctrl+c, ctrl+x, ctrl+v).

To make C&P available to the pltbrowser plugin I added an API to the gtkui plugin:
https://github.com/Alexey-Yakovenko/deadbeef/compare/master...cboxdoerfer:copy_paste?expand=1#diff-31721c7ad41dcfea0c8abdd8072c53e9R263

Hope I can get any feedback on that. I also thought about making a proper GObject, e.g. DdbClipboard, but found that to much overhead, with little benefit.

Alexey-Yakovenko commented on plugins/gtkui/clipboard.c:
```diff
@@ -0,0 +1,548 @@
+   since we need to decode the tracks all over again.
+
+   As soon as we have a database we could improve that, because tracks would
+   be bound to the database and not the playlist and therefore be more persitent. */
```
I'm not sure if I understand it correctly, but the tracks (playItems) are reference counted, and if you do `pl_item_ref` before storing a pointer -- they are guaranteed to persist, then you can do `pl_item_unref` before quitting the player and clearing clipboard..

cboxdoerfer replied:
... of course, you're right and I'm stupid. :D No idea why I haven't thought of that myself. Well at least it's easy to fix.

Alexey-Yakovenko commented on plugins/gtkui/clipboard.c:
```diff
@@ -0,0 +1,548 @@
+}
+
+static void
+clipboard_clear_clipboard_data (GtkClipboard *clipboard,
```
Is this the same as `clipboard_free_clipboard_data`?

cboxdoerfer replied:
Yes thx, they started out different but apparently became the same over time. Will get rid of one of them.

Alexey-Yakovenko commented:
`clipboard_free_clipboard_data` - would look better if it's called `clipboard_free_data`, or even `clipboard_free` (depends).

cboxdoerfer commented:
@Alexey-Yakovenko, what do you think of the gtkui plugin api? Do you think it's ok to useDDB_ACTION_CTX_* there?

Alexey-Yakovenko commented:
My understanding of the code is not yet good enough to tell whether it's good or bad. Let's say it's ok for now, we can change it any time.

cboxdoerfer commented:
Fixed the issues. I also added a ref counter for the clipboard data to make sure we do proper clean up at exit. From my understanding of the API GTK+ is supposed to do that, but it doesn't work reliably, so we do it manually if GTK+ fails.
